src/Uno.UI/UI/Xaml/Controls/Layouter/Layouter.Android.cs

```diff
@@ -58,6 +58,8 @@ protected Size MeasureChildOverride(View view, Size slotSize)
 			var widthSpec = ViewHelper.SpecFromLogicalSize(slotSize.Width);
 			var heightSpec = ViewHelper.SpecFromLogicalSize(slotSize.Height);
 
+			view.ForceLayout(); // Bypass Android cache, to ensure the Child's Measure() is actually invoked.
```

Comment:
This is a pretty aggressive approach, which will negatively impact performance, potentially severely. It basically makes it so that the entire visual tree will remeasure whenever any view is invalidated.

Normally, the native Android cache should be invalidated whenever the available dimensions change, or when the child itself calls `RequestLayout()`. Can you say why that's not working or not sufficient here? Is there a way to apply this 'brute force' approach only in a more specific case?

Reply:
That's the main difference between `.ForceLayout()` and `.RequestLayout()`.  The first one will _only_ bypass the cache of the control to force the `.Measure()` to be actually invoked. The `.RequestLayout()` will invalidate all the tree.

Reply:
But you're calling `ForceLayout()` recursively!

Reply:
Consider the following tree:

```
    A
  | | |
  v v v
  B C D
    |
    v
    E
```

1. Call `RequestLayout()` on E. This will invalidate E, C, and A, and prompt the window to schedule a layout pass.

2. A will be measured. (Refer to [this diagram](https://platform.uno/docs/articles/uno-development/Uno-UI-Layouting-Android.html) for more details.) It will measure its children, B, C, and D. Normally it should see that B and D are unchanged and skip them. This is implicitly handled by the native caching.

_But_ now you're calling `ForceLayout()` on B and D, so they'll be remeasured. And recursively, in the same way, the entire subtrees under them. So a huge chunk of the visual tree will be remeasured that doesn't need to be. This is why this is a significant performance concern.

Reply:
There's measure caching integrated in the layouter, but I'm wondering if the issue here is more around the fact that the TextBlock is not invalidating properly based on changes of the arrange values.

Reply:
IIRC for iOS Uno does some measure caching, for Android we mostly rely on the native caching.

Reply:
@davidjohnoliver A better fix has been found. The invalidation will be done only when we know it's required.

Reply:
It's less extreme, but it could still be a performance hit in the right (wrong) circumstances, namely a view at the root of a large subtree that consistently returns a larger desired size than its available bounds.

And I still wonder if it's really necessary or if it's masking a deeper problem that could/should be fixed.

Reply:
@davidjohnoliver @jeromelaban 
Ok, here's my final fix ;-)